subprojects/code-quality/src/main/groovy/org/gradle/api/plugins/quality/CodeNarc.java

```diff
@@ -62,9 +72,19 @@ public class CodeNarc extends SourceTask implements VerificationTask, Reporting<
 
     private boolean ignoreFailures;
 
+    private final Property<JavaLauncher> javaLauncher;
+
     public CodeNarc() {
         reports = getObjectFactory().newInstance(CodeNarcReportsImpl.class, this);
         compilationClasspath = getProject().files();
+        // Set default JavaLauncher to current JVM in case
+        // CodeNarcPlugin that sets Java launcher convention is not applied
+        this.javaLauncher = getCurrentJvmLauncher();
```

Comment:
It reads weird to have the constructor call what looks to be a getter.  Maybe rename this to `configureFromCurrentJvmLauncher()` or something.  I'd also consider making that method `static` to make it clear it can't rely on instance state, since it's being called from the constructor.